### PR TITLE
Standardize code formatting in repository

### LINKED ISSUE #1754: Lint Markdown/JavaScript files in repository
## Why
We want to standardize the linting of the Markdown (and JavaScript) files in the repository

## What
Enforce linting of the Markdown (and JavaScript) files in the repository

## How

- [x] Add a configuration file for `prettier`?
- [x] Add a command to lint the files in the repository?
- [x] Enforce the linting in the CI

### PR DESCRIPTION
**Pull request by dlachaume**
## Content
This PR includes a global configuration for the code formatting strategy.

This configuration is managed by [Prettier](https://prettier.io/) and can be configured in the `.prettierc` file located at the root of the repository with [options](https://prettier.io/docs/en/options.html).

The Makefiles of `mithril-client-wasm/www/`, `mithril-client-wasm/www-test/`, and `docs/website/` have been extended with `make lint` and `make format` commands.

`make lint` can also be executed from the root of the repository to facilitate usage and CI checks for the entire repository.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [ ] Crates versions are updated (if relevant)
  - [x] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [x] Update README file (if relevant)
  - [x] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)
Closes #1754

### CODE REVIEW

File: .prettierrc
Comment (jpraynaud) [inline .prettierrc:1]: Maybe we could have the formatting rules more explicit for tab width, single/double quotes, ... and also for each type of file (there is no markdown file entry)

**Review by sfauvel — APPROVED** [review]
LGTM

File: docs/website/root/manual/getting-started/run-signer-node.md [hunk 1]
```diff
@@ -263,43 +269,44 @@ sudo mv mithril-signer /opt/mithril
...
+- \***\*YOUR_OPERATIONAL_CERTIFICATE_PATH\*\***: `/cardano/keys/node.cert`
+- \***\*YOUR_CARDANO_NETWORK\*\***: `preprod`
+- \***\*YOUR_AGGREGATOR_ENDPOINT\*\***: `https://aggregator.release-preprod.api.mithril.network/aggregator`
+- \***\*YOUR_ERA_READER_ADAPTER_TYPE\*\***: `cardano-chain`
+- \***\*YOUR_ERA_READER_ADAPTER_PARAMS\*\***: `{"address": "addr_test1qpkyv2ws0deszm67t840sdnruqgr492n80g3y96xw3p2ksk6suj5musy6w8lsg3yjd09cnpgctc2qh386rtxphxt248qr0npnx", "verification_key": "5b35352c3232382c3134342c38372c3133382c3133362c34382c382c31342c3138372c38352c3134382c39372c3233322c3235352c3232392c33382c3234342c3234372c3230342c3139382c31332c33312c3232322c32352c3136342c35322c3130322c39312c3132302c3230382c3134375d"}`
+- \***\*YOUR_RELAY_ENDPOINT\*\***: `192.168.1.50`
+- \***\*YOUR_RELAY_LISTENING_PORT\*\***: `3132`
+- \***\*YOUR_BLOCK_PRODUCER_INTERNAL_IP\*\***: `192.168.1.75`
+- \***\*YOUR_SIGNER_LOGS_PATH\*\***: `/var/log/syslog`
+- \***\*YOUR_PARTY_ID\*\***: `pool1hp72sauk0g0yqm4dzllz0pz6j93gewhllkzphn4hykkfmne43y`
```
Comment (jpraynaud) [inline docs/website/root/manual/getting-started/run-signer-node.md:302]: I'm not sure that this doesn't change the way it is displayed on the website :thinking:

File: flake.lock
Comment (jpraynaud) [inline flake.lock:1]: Maybe we can skip formatting lock files from flake and maybe also from npm?

File: mithril-test-lab/mithril-devnet/configuration/babbage/alonzo-babbage-test-genesis.json
Comment (jpraynaud) [inline mithril-test-lab/mithril-devnet/configuration/babbage/alonzo-babbage-test-genesis.json:1]: Can you skip the formatting of the files in `mithril-test-lab/mithril-devnet/configuration`?

**Review by jpraynaud — COMMENTED** [review]
Well done :+1: 

Few remarks and comments before it's good to merge.

File: .prettierrc
Comment (dlachaume) [reply .prettierrc:1]: Indeed, I see. These are the default values being used except for `.js` and `.yaml`, but it’s not enough explicit.
To be sure I understand correctly, you prefer that I clearly indicate the default values for `tabWidth` and `singleQuote` options and by file type, even if it’s redundant?

File: Makefile [hunk 2]
```diff
@@ -2,10 +2,17 @@ COMPONENTS = mithril-common mithril-stm mithril-aggregator mithril-client mithri
...
+$(filter-out $(NON_COMPONENT_GOALS),$(GOALS)): $(COMPONENTS)
 
 $(COMPONENTS):
 	$(MAKE) -C $@ $(MAKECMDGOALS)
+
+lint:
+	prettier --check .
+
+format:
+	prettier --write .
```
Comment (Alenar) [inline Makefile:18]:
```suggestion
check-format:
	@echo 'Note: Rust is not checked by this recipe, use `make check`'
	@which prettier >/dev/null || echo 'It seems 'prettier' is not installed or not in the path. see https://prettier.io/docs/en/install';
	prettier --check .

format:
	@which prettier >/dev/null || echo 'It seems 'prettier' is not installed or not in the path. see https://prettier.io/docs/en/install';
	prettier --write .
```

File: flake.lock
Comment (dlachaume) [reply flake.lock:1]: Done ✅

File: mithril-test-lab/mithril-devnet/configuration/babbage/alonzo-babbage-test-genesis.json
Comment (dlachaume) [reply mithril-test-lab/mithril-devnet/configuration/babbage/alonzo-babbage-test-genesis.json:1]: Done ✅

**Review by Alenar — APPROVED** [review]

File: docs/website/root/manual/getting-started/run-signer-node.md (hunk 1 above)
Comment (dlachaume) [reply docs/website/root/manual/getting-started/run-signer-node.md]: Indeed, fixed ✅

**Review by jpraynaud — APPROVED** [review]
LGTM :rocket:

**Pull request merged**